Add --custom-log-dict CLI option

isobelhooper opened the pull request:
This points at an import path for a dictionary (e.g. myproject.logging.CONFIG_DICT) that Arq can configure its logging with instead of using the default configuration.

This would be useful for e.g. configuring Arq logging output to use [python-json-logger](https://github.com/madzak/python-json-logger).

We use this in a fork of Arq for a project I'm working on, and I wanted to contribute this back upstream. I'm happy to make any changes that are needed.

samuelcolvin commented on arq/cli.py:
```diff
@@ -28,15 +30,24 @@
+            config_path, config_dict = custom_log_dict.rsplit('.', maxsplit=1)
+            log_config = getattr(importlib.import_module(config_path), config_dict)
+        except (TypeError, AttributeError):
+            log_config = default_log_config(verbose)
```
we should log (probably just print to sterr is easiest) a warning in this case with details of what went wrong.

Or we could just raise an error.

isobelhooper replied:
Raising an ImportError from `import_string` seems like it would work okay - I'll do that.

samuelcolvin reviewed (commented):
I'm okay with this, needs also:
* a test
* to be rebased to master - should fix build

samuelcolvin commented:
you've got an error in CI on py3.10. You (or I) will also need to pin jinja to make docs build.

samuelcolvin commented:
update, I've pinned jinja in master.

isobelhooper commented:
This is now rebased on current master, and I've also updated the version of black to deal with an incompatibility with certain Click versions. Let me know if there's anything else you'd like me to change in this.

samuelcolvin commented:
thanks so much.
